ginga/canvas/types/astro.py

```diff
@@ -247,7 +247,25 @@ def draw(self, viewer):
             self.draw_caps(cr, self.cap, ((cx2, cy1), ))
 
 
-class Compass(OnePointOneRadiusMixin, CanvasObjectBase):
+class Ruler(RulerP):
+
+    @classmethod
+    def idraw(cls, canvas, cxt):
+        return cls(cxt.start_x, cxt.start_y, cxt.x, cxt.y, **cxt.drawparams)
+
+    def __init__(self, x1, y1, x2, y2, color='green', color2='skyblue',
+                 alpha=1.0, linewidth=1, linestyle='solid',
+                 showcap=True, showplumb=True, showends=False, units='arcmin',
+                 font='Sans Serif', fontsize=None, **kwdargs):
+        RulerP.__init__(self, (x1, y1), (x2, y2), color=color, color2=color2,
```

Comment:
Why not use `super`?

Reply:
It might work, but the parent class is derived from mixin classes and uses explicit class calls for it's constructor.  So I'm just trying to be consistent here.